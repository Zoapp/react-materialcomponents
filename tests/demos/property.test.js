import React from "react";
import renderer from "react-test-renderer";
import Property from "demos/components/property";

describe("demos/components/property", () => {
  it("renders itself correctly", () => {
    const componentSingle = renderer
      .create(<Property name="Button" />)
      .toJSON();
    expect(componentSingle).toMatchSnapshot();
    const componentMultiple = renderer
      .create(<Property name="Card" />)
      .toJSON();
    expect(componentMultiple).toMatchSnapshot();
  });
});
