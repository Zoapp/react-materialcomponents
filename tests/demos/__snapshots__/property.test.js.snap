// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`demos/components/property renders itself correctly 1`] = `null`;

exports[`demos/components/property renders itself correctly 2`] = `
Array [
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
        style={
          Object {
            "fontSize": "20px",
            "fontWeight": "bold",
          }
        }
      >
        Card Properties
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
      >
        Name
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Default
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Type
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        mdcElement
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        MDC_CARD
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        children
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        node
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        title
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        subtitle
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        largeTitle
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        horizontalBlock
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
  </div>,
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
        style={
          Object {
            "fontSize": "20px",
            "fontWeight": "bold",
          }
        }
      >
        Card Actions Properties
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
      >
        Name
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Default
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Type
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        mdcElement
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        MDC_CARD_ACTIONS
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        children
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        node
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        vertical
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
  </div>,
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
        style={
          Object {
            "fontSize": "20px",
            "fontWeight": "bold",
          }
        }
      >
        Card Title Properties
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
      >
        Name
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Default
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Type
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        mdcElement
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        MDC_CARD_TITLE
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        children
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        node
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        large
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
  </div>,
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
        style={
          Object {
            "fontSize": "20px",
            "fontWeight": "bold",
          }
        }
      >
        Card Subtitle Properties
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
      >
        Name
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Default
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Type
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        mdcElement
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        MDC_CARD_SUBTITLE
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        children
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        node
      </div>
    </div>
  </div>,
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
        style={
          Object {
            "fontSize": "20px",
            "fontWeight": "bold",
          }
        }
      >
        Card Media Properties
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
      >
        Name
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Default
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Type
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        mdcElement
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        MDC_CARD_MEDIA
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        children
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        node
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        title
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        subTitle
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        largeTitle
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        horizontalBlock
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
  </div>,
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
        style={
          Object {
            "fontSize": "20px",
            "fontWeight": "bold",
          }
        }
      >
        Card Text Properties
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
      >
        Name
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Default
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Type
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        mdcElement
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        MDC_CARD_TEXT
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        children
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        node
      </div>
    </div>
  </div>,
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
        style={
          Object {
            "fontSize": "20px",
            "fontWeight": "bold",
          }
        }
      >
        Card Primary Properties
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
      >
        Name
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Default
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Type
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        mdcElement
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        MDC_CARD_PRIMARY
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        children
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        node
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        title
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        subtitle
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        largeTitle
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        isTitlingSecond
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
  </div>,
]
`;
